### PR TITLE
Fix pixman macOS build and add missing build deps

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

**Review by wdconinc — APPROVED** [review]
This works fine on my system.

Unrelated Gitlab CI issue fixed in https://github.com/spack/spack/pull/36985, rebase should pass it.

File: var/spack/repos/builtin/packages/pixman/package.py [hunk 1]
```diff
@@ -25,6 +25,8 @@ class Pixman(AutotoolsPackage):
     version("0.32.6", sha256="3dfed13b8060eadabf0a4945c7045b7793cc7e3e910e748a8bb0f0dc3e794904")
 
     depends_on("pkgconfig", type="build")
+    depends_on("flex", type="build")
+    depends_on("bison@3:", type="build")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/pixman/package.py:29]: Do these dependencies apply to all of the versions or only for those from some release on?

Comment (sethrj) [reply var/spack/repos/builtin/packages/pixman/package.py:29]: As far as I know, all of them? It's an iteration in the right direction; if someone discovers an incompatibility with an older version we can improve the constraints then I think

**Comment by sethrj** [comment]
Ping @tldahlgren

**Review by tldahlgren — APPROVED** [review]
LGTM

**Comment by wdconinc** [comment]
~~I can't merge this and I have already approved. What should I do?~~ Nevermind, looks good now :-)

**Comment by tldahlgren** [comment]
> I can't merge this and I have already approved. What should I do?

You commented mid-operation (while my attention was momentarily averted).

**Pull request merged**